### DISCUSSION TITLE
all setup up significations on solax ha integration.

### DISCUSSION THREAD

**Discussion by gazzzzton** [wills106/homeassistant-solax-modbus#218] [Q&A]
hi,
congrats for all your work.
i use solax intégration for my solax x1 hybrid gen4 whith pylontech battery and i'm so confuse about all setups i can use.
someone could define me all setup and it signification.
per exemple, my solax on self use mode switch between normal mode and stand-by and i don't now why it do this ...
sorry for my bad english.

**Comment by wills106** [comment]
Have you looked in the Wiki? There are overviews for the Gen2 / Gen3 and a separate one for the Gen4

**Reply by gazzzzton** [reply]
yes i've seen that and my setup are ok for this section.
my question is about the setups on the intégration.
there are more setup on ha intégration and i don't know there signification.
for exemple i don't know the signification of:
- external génération (i think it's solar panel)
- maximum per day on
- maximum per on signal
- selfuse nightcharge upper soc
- switch off soc
- switch on soc
- work mode
- selfuse backup soc

could you help me ? 
thanks in advance

**Comment by gazzzzton** [comment]
![Screenshot_2022-11-28-09-03-20-015_io homeassistant companion android](https://user-images.githubusercontent.com/63874950/204227602-7d39a6c8-e814-4b05-8f6d-3de52f109dac.jpg)

charge is disable and discharge enable
you could see on my picture the problem along the night. so problem the day when solar is present...

**Reply by wills106** [reply]
It could be in one of these modes.

https://github.com/wills106/homeassistant-solax-modbus/blob/09da6e24aa4428183c27fecd40083c149ddc0951/custom_components/solax_modbus/plugin_solax.py#L2417-L2435

**Comment by gazzzzton** [comment]
yes all night it switches at regular intervals between "normal mode" (battery) and "on standby" (grid) while I would like it to remain in "normal mode" all the time as long as my battery soc is greater than 21 %.
What settings do you think I should touch to prevent it from permanently switching from one to the other?
